sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLogSuite.scala

```diff
@@ -60,4 +60,39 @@ class OffsetSeqLogSuite extends SparkFunSuite with SharedSQLContext {
         Array(0 -> batch0Serialized, 1 -> batch1Serialized))
     }
   }
+
+  test("read Spark 2.1.0 log format") {
+    val (batchId, offsetSeq) = readFromResource("offset-log-version-2.1.0")
+    assert(batchId === 16)
+    assert(offsetSeq ===
+      OffsetSeq(
+        offsets = Seq(Some(SerializedOffset(
+          """
+            |{"kafka-topic":{"23":0,"8":1,"17":1,"11":1,"20":0,"2":6,"5":2,"14":0,"4":4,"13":1,
```

Comment:
We should add a test for Kafka source to check if we can convert json to KafkaSourceOffset. This doesn't check that.